x/ecocredit/client/testsuite/tx.go

```diff
@@ -1373,3 +1393,417 @@ func (s *IntegrationTestSuite) TestTxUpdateIssuers() {
 		})
 	}
 }
+
+func (s *IntegrationTestSuite) TestTxSell() {
+	val0 := s.network.Validators[0]
+	clientCtx := val0.ClientCtx
+
+	testCases := []struct {
+		name      string
+		args      []string
+		expErr    bool
+		expErrMsg string
+		expOrder  *ecocredit.SellOrder
+	}{
+		{
+			name:      "missing args",
+			args:      []string{},
+			expErr:    true,
+			expErrMsg: "accepts 1 arg(s), received 0",
+		},
+		{
+			name:      "too many args",
+			args:      []string{"foo", "bar"},
+			expErr:    true,
+			expErrMsg: "accepts 1 arg(s), received 2",
+		},
+		{
+			name: "missing batch denom",
+			args: append(
+				[]string{
+					"[{quantity: \"5\", ask_price: \"100regen\", disable_auto_retire: false}]",
```

Comment:
might be a little cleaner/straightforward to just json.Marshal this using the `encoding/json` package.

i dont think this works for something like MsgSell as it uses coin for AskPrice, but we could just make a mirrored test struct changing that field to a string. 

not sure if that change would be all that valuable, so interested on seeing thoughts on this

Reply:
Hmmm.... I might be misunderstanding what you're proposing but I don't think we need to create a separate struct and marshal it for the arguments within the test cases. I was following `Send`, `Cancel`, and `Retire` as examples. I think the argument here as a string is important as this is the same format we are expecting users to provide.